stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/core.py

```diff
@@ -51,23 +41,14 @@ class CoreCrudClient(BaseCoreClient):
     )
     settings = ElasticsearchSettings()
     client = settings.create_client
+    database = CoreDatabaseLogic()
```

Comment:
It looks like this `database` attribute completely abstracts away access to the `client` and `settings`, is that correct? If so, the `client` and `settings` attributes should be removed.

Also, is there any reason not to just call it `Database` instead of `CoreDatabaseLogic`?

Reply:
True - good catch. I was going to have a TransactionsDatabaseLogic but it could all be one class or have a better name too.

Reply:
+1 for just having one class for both.

perhaps DatabaseClient? I assume the longer-term goal here is to have ElasticsearchDatabaseClient and OpensearchDatabaseClient, that abstract away the small differences between the two.

Reply:
I could also have a MongoClient outside of the repo that I could maybe inject when I wanted to and not have to keep a full fast-stac-api backend up to date.